### PR TITLE
docs(blog): Add post about Gatsby Themes and Theme UI

### PR DESCRIPTION
**Pull request by LekoArts**
Add my post as requested by @lindaleebumblebee 	
https://www.lekoarts.de/en/blog/how-i-used-theme-ui-to-build-my-gatsby-themes-library

### CODE REVIEW

File: docs/blog/2019-07-26-how-i-used-theme-ui-to-build-my-gatsby-themes-library/index.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+canonicalLink: "https://www.lekoarts.de/en/blog/how-i-used-theme-ui-to-build-my-gatsby-themes-library"
+publishedAt: lekoarts.de
+tags:
+  - themes
+  - theme-ui
+---
+
+_This blog post was originally posted on [LekoArts' Blog](https://www.lekoarts.de/en/blog/how-i-used-theme-ui-to-build-my-gatsby-themes-library)_
+
+Building [Themes](https://www.gatsbyjs.org/blog/2019-07-03-announcing-stable-release-gatsby-themes/) for Gatsby is fun and a great way to share projects with the Open Source community. In the past I mainly created starters which I added to the [starter overview](https://www.gatsbyjs.org/starters/?v=2) on Gatsbyjs.org and called it a day. While working on my first starter to theme conversion I realized that I don't have a great overview or marketing site for my starters — I wanted to change that for my upcoming themes! Hence I created [themes.lekoarts.de](https://themes.lekoarts.de). It currently shows my available themes and their respective links to GitHub and demo website. Moreover I plan to send out a newsletter every month so be sure to subscribe :)
```
Comment (amberleyromo) [inline docs/blog/2019-07-26-how-i-used-theme-ui-to-build-my-gatsby-themes-library/index.md]:
```suggestion
Building [themes](https://www.gatsbyjs.org/blog/2019-07-03-announcing-stable-release-gatsby-themes/) for Gatsby is fun and a great way to share projects with the open source community. In the past I mainly created starters which I added to the [starter library](https://www.gatsbyjs.org/starters/?v=2) on Gatsbyjs.org and called it a day. While working on my first starter to theme conversion I realized that I don't have a great overview or marketing site for my starters — I wanted to change that for my upcoming themes! Hence I created [themes.lekoarts.de](https://themes.lekoarts.de). It currently shows my available themes and their respective links to GitHub and demo website. Moreover I plan to send out a newsletter every month so be sure to subscribe :)
```

File: docs/blog/2019-07-26-how-i-used-theme-ui-to-build-my-gatsby-themes-library/index.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+With this blogpost I want to describe the process of creating my Gatsby Themes library website with [Theme UI](https://theme-ui.com/). I won't go too deep into technical details so please don't expect this to be a tutorial.
```
Comment (amberleyromo) [inline docs/blog/2019-07-26-how-i-used-theme-ui-to-build-my-gatsby-themes-library/index.md]:
```suggestion
With this blogpost I want to describe the process of creating my Gatsby themes library website with [Theme UI](https://theme-ui.com/). I won't go too deep into technical details so please don't expect this to be a tutorial.
```

File: docs/blog/2019-07-26-how-i-used-theme-ui-to-build-my-gatsby-themes-library/index.md [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+## Design
+
+Nearly every time before I start coding a project I start with the design in Figma. I'm still faster with prototyping with Figma than in code with React (probably due to my design background). I started looking at Dribbble and Behance to find some inspiration as I had a light website with playful elements, a **stunning** hero section and otherwise minimal design in mind.
+
+<figure>
+<img src="./images/behance-dribbble-inspiration.png" alt="Screenshot of Figma showing different screenshots from Dribbble, Behance and Refactoring UI for inspiration" />
+<figcaption>Inspirational screenshots</figcaption>
+</figure>
+
+I really liked the prominent and actionable hero section of the Refactoring UI website, the shapes and colors from the other designs. Speaking of Refactoring UI: For the design (and later for coding) I used the TailwindCSS colors and styles.
```
Comment (amberleyromo) [inline docs/blog/2019-07-26-how-i-used-theme-ui-to-build-my-gatsby-themes-library/index.md]:
```suggestion
I really liked the prominent and actionable hero section of the [Refactoring UI website](https://refactoringui.com/), the shapes and colors from the other designs. Speaking of Refactoring UI, for the design (and later for coding) I used the TailwindCSS colors and styles.
```

File: docs/blog/2019-07-26-how-i-used-theme-ui-to-build-my-gatsby-themes-library/index.md [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+
+The design you see live on the website now is pretty much the design I created in Figma — which speaks for the ease of use of a Figma (+ Tailwind styles) => Theme UI (+ Tailwind preset) workflow!
```
Comment (amberleyromo) [inline docs/blog/2019-07-26-how-i-used-theme-ui-to-build-my-gatsby-themes-library/index.md]:
```suggestion
The design you see live on the website now is pretty much the design I created in Figma — which speaks to the ease of use of a Figma (+ Tailwind styles) => Theme UI (+ Tailwind preset) workflow!
```

File: docs/blog/2019-07-26-how-i-used-theme-ui-to-build-my-gatsby-themes-library/index.md [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+As mentioned I also wanted a rad hero section, nothing boring but also not something that is overused (and potentially can also slow down your website) like particles.js. Since I'm pretty proficient with Cinema4D and 3D in general I just tried to visualize "Gatsby Themes" with 3D models. I drew some inspiration from [Kyle Gill's illustrations](https://kylegill.com/blog/2019-04-22-a-simple-guide-to-gatsby-themes/index). I rendered the scene with Arnold, modified it a bit with Photoshop and et voilá: You can see the finished result on my social preview image now.
+
+<figure>
+<img src="./images/gatsby-themes-3d-illustration.png" alt="3D Rendering showing a little factory receiving input from four objects." />
+<figcaption>Cinema4D Rendering of a "Gatsby Themes factory"</figcaption>
+</figure>
+
+## Code
+
+The website lives inside the [www](https://github.com/LekoArts/gatsby-themes/tree/master/www) folder of the monorepo and thus is already setup to be used with TypeScript. You can have a look at the source code to see how it's done (as going through the complete code would be a bit boring...). A follow-up blogpost will explain how I created the 3D element in the hero section.
```
Comment (amberleyromo) [inline docs/blog/2019-07-26-how-i-used-theme-ui-to-build-my-gatsby-themes-library/index.md]:
```suggestion
The website lives inside the [www](https://github.com/LekoArts/gatsby-themes/tree/master/www) folder of the monorepo and thus is already setup to be used with TypeScript. You can have a look at the source code to see how it's done (as going through the complete code would be a bit boring...).
```

File: docs/blog/2019-07-26-how-i-used-theme-ui-to-build-my-gatsby-themes-library/index.md [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+<figure>
+<img src="./images/figma-theme-ui-handoff.png" alt="Screenshot showing both Figma and VS Code. The value used in Figma can easily get transferred to VS Code and Theme UI." />
+<figcaption>Using Tailwind styles in both Figma and Theme UI is the best!</figcaption>
+</figure>
+
+It's great to have the same design tokens in all places. I chose the color **Indigo.600** and later in Theme UI I just had to use `colors.indigo[6]`.
+
+### Variants
+
+The bottom view of VS Code shows a snippet of the button variants — variants are a Theme UI feature (see [documentation](https://theme-ui.com/guides/variants)). I used them for the buttons and gradients. The latter looks something like this:
```
Comment (amberleyromo) [inline docs/blog/2019-07-26-how-i-used-theme-ui-to-build-my-gatsby-themes-library/index.md]: > The bottom view of VS Code

Where is this? Am I missing it?

File: docs/blog/2019-07-26-how-i-used-theme-ui-to-build-my-gatsby-themes-library/index.md [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+  <div
+    sx={{
+      variant: `gradients.${color}`,
+    }}
+    className={className}
+  />
+)
+```
+
+Little tip here: When you pass `className` down, you can style the component individually when using it.
```
Comment (amberleyromo) [inline docs/blog/2019-07-26-how-i-used-theme-ui-to-build-my-gatsby-themes-library/index.md]:
```suggestion
> Little tip here: When you pass `className` down, you can style the component individually when using it.
```

File: docs/blog/2019-07-26-how-i-used-theme-ui-to-build-my-gatsby-themes-library/index.md [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+    )
+  })}
+</Container>
+```
+
+Setting the offset values to either _left_ or _right_ is decided by _isEven_ which is also used to change the order of the items. Passing in the arrays into the props works like on the rest of the site and hence the responsive styling!
+
+## Conclusion
+
+All in all Theme UI helped me a lot to get this done quickly. Starting at the Design to Code Handoff (by using the same design tokens) and ending at using the breakpoint syntax in the GraphQL layer of Gatsby to individually style theme entries. Of course Theme UI offers you all the things it says on its own website: Ergonomic, Themeable, Contraint-based. And in the case of Gatsby Themes: Also shadowable.
```
Comment (amberleyromo) [inline docs/blog/2019-07-26-how-i-used-theme-ui-to-build-my-gatsby-themes-library/index.md]:
```suggestion
All in all Theme UI helped me a lot to get this done quickly. From the design phase, to the code handoff phase (by using the same design tokens) to finally using the breakpoint syntax in the GraphQL layer of Gatsby to individually style theme entries. Of course Theme UI offers you all the things it says on its own website: "Ergonomic, Themeable, Contraint-based". And in the case of Gatsby themes: Also shadowable.
```

Comment (amberleyromo) [reply docs/blog/2019-07-26-how-i-used-theme-ui-to-build-my-gatsby-themes-library/index.md]: > All in all Theme UI helped me a lot to get this done quickly. Starting at the Design to Code Handoff (by using the same design tokens) and ending at using the breakpoint syntax in the GraphQL layer of Gatsby to individually style theme entries.

I'm not _entirely_ sure I read this sentence correctly, does my suggestion make sense?

File: docs/blog/2019-07-26-how-i-used-theme-ui-to-build-my-gatsby-themes-library/index.md (hunk 6 above)
Comment (LekoArts) [reply docs/blog/2019-07-26-how-i-used-theme-ui-to-build-my-gatsby-themes-library/index.md]: I'm referring to the figma-theme-ui-handoff screenshot above. Seems like it needs some clarification

File: docs/blog/2019-07-26-how-i-used-theme-ui-to-build-my-gatsby-themes-library/index.md (hunk 8 above)
Comment (LekoArts) [reply docs/blog/2019-07-26-how-i-used-theme-ui-to-build-my-gatsby-themes-library/index.md]: Yeah, I realize now that it works like that in german but not english 😅 It should summarize the process in a short manner and I think your suggestion also does that

File: docs/blog/2019-07-26-how-i-used-theme-ui-to-build-my-gatsby-themes-library/index.md (hunk 6 above)
Comment (amberleyromo) [reply docs/blog/2019-07-26-how-i-used-theme-ui-to-build-my-gatsby-themes-library/index.md]: Perfect, thanks!

**Pull request merged**